src/CloudSecrets/SecretsAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.IO.Abstractions;
+using System.Linq;
+using System.Threading;
+using SonarLint.Secrets.DotNet;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Analysis;
+
+namespace SonarLint.VisualStudio.CloudSecrets
+{
+    [Export(typeof(IAnalyzer))]
+    internal class SecretsAnalyzer : IAnalyzer
+    {
+        private readonly IEnumerable<ISecretDetector> secretDetectors;
+        private readonly IAnalysisStatusNotifier analysisStatusNotifier;
+        private readonly IFileSystem fileSystem;
+
+        [ImportingConstructor]
+        public SecretsAnalyzer([ImportMany] IEnumerable<ISecretDetector> secretDetectors, IAnalysisStatusNotifier analysisStatusNotifier)
+            : this(secretDetectors, analysisStatusNotifier, new FileSystem())
+        {
+        }
+
+        internal SecretsAnalyzer(IEnumerable<ISecretDetector> secretDetectors,
+            IAnalysisStatusNotifier analysisStatusNotifier,
+            IFileSystem fileSystem)
+        {
+            this.secretDetectors = secretDetectors;
+            this.analysisStatusNotifier = analysisStatusNotifier;
+            this.fileSystem = fileSystem;
+        }
+
+        public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
+        {
+            return true;
+        }
+
+        public void ExecuteAnalysis(string filePath,
+            string charset,
+            IEnumerable<AnalysisLanguage> detectedLanguages,
+            IIssueConsumer consumer,
+            IAnalyzerOptions analyzerOptions,
+            CancellationToken cancellationToken)
+        {
+            analysisStatusNotifier.AnalysisStarted(filePath);
+
+            try
+            {
+                var stopwatch = Stopwatch.StartNew();
+
+                var fileContent = fileSystem.File.ReadAllText(filePath);
+
+                // todo: pass rules configuration if should run the detector? or should each detector expose metadata?
```

Comment:
will be handled in #2679

Reply:
Let's discuss this offline. FYI my current thinking is as follows:

I was thinking that the detectors would expose a rule key, and we'd do the check here.

The considerations are the similar for the question about cancellation below - we could make the detectors as simple as possible, and have more of the logic here, or we could push more of the responsibility down to the individual detectors.

Currently, I think we should make the detectors as simple as possible.
* if we want to pass the rule configuration to the detectors then we need a shared API that the secrets assembly and SLVS can both reference, which we don't currently have.
* the detector rules will be direct ports of the equivalent Java code, and. I think keeping the code as minimal as possible would make it simpler to maintain i.e. don't add responsibilities in the .NET code that aren't in the Java code.

We could pass the cancellation token if we think the code in each detector will take long enough to run  and executes in multiple steps i.e. so it is worth checking for cancellation between each step.
At the moment it looks like the detectors will only have one main step, so it's probably not worth passing the token.